devito/at_controller.py

```diff
@@ -0,0 +1,344 @@
+        """
+
+        log_file_path = os.path.join(self._at_parent_folder, "log.txt")  # Writes STDOUT in here
+        log = open(log_file_path, 'w')
+
+        # auto tuning command. -i for source of at -o for destination of at result
+        cmd = "%s -i %s -o %s" % (self._isat_py_path, os.path.join(self._at_parent_folder, self._at_src_dir, ),
+                                  os.path.join(self._at_parent_folder, self._at_dst_dir))
+        process = subprocess.Popen(cmd, shell=True, universal_newlines=True,
+                                   stdout=log, stdin=subprocess.PIPE, stderr=subprocess.STDOUT)
+        process.stdin.write('y')  # isat asks whether we want to continue
+        process.communicate()[0]
+
+        log.close()
+
+        if process.returncode != 0:
+            print("Isat.py return code != 0. Check %s for errors" % log_file_path)
+            raise ValueError()
+
+    def _extract_report_info(self):
+        """Copies report over to auto-tune-reports dir and extracts best block size from it
+
+        Raises:
+            ValueError: If auto tuning report is not found
+        """
+
+        global reports_folder_path
+
+        # prep report dir if does not exist
+        if not os.path.isdir(reports_folder_path):
+            os.mkdir(reports_folder_path)
+
+        at_report_path = os.path.join(os.path.join(self._at_parent_folder, self._at_dst_dir), "isat-report.txt")
+
+        if not os.path.isfile(at_report_path):  # checks if report file exist
+            print "%s not found" % at_report_path
+            raise ValueError()
+
+        # extracts the best block size
+        self._extract_best_block_size(at_report_path)
+
+        # note if running on windows change into back slash
+        split = self._at_parent_folder.split('/')
+        report_name_ref = split[len(split) - 1]  # get at parent folder name which is used for naming reports
+
+        # copy file for ref
+        shutil.copyfile(at_report_path, os.path.join(reports_folder_path, "%s-at-report.txt" % report_name_ref))
+
+    def _extract_best_block_size(self, at_report_path):
+        """Extracts the best best block size from at report and writes it to  the final report file
+
+        Args:
+            at_report_path (str): full path to a at report
+
+        Raises:
+            ValueError: If failed to extract block size. Can't open report files, can't write to them. etc
+        """
+        global final_report_path
+        try:
+            # Gets the required info from  at report
+            with open(at_report_path, 'r') as at_report:
+                # if all these keywords are in the line then next line is the one we want
+                keyword_lst = ["Rank", "Value", "Time(secs)"]
+
+                lines = at_report.readlines()
+                for i in range(0, len(lines)):  # loops through all the lines trying to find the best
+                    if all(keyword in lines[i] for keyword in keyword_lst):
```

Comment:
Since we are trying to be "pythonic" in this project, I would disagree with using C# style syntax for something that is easily achievable by an equivalent pythonic syntax.